NYPLCardCreator/BusinessLogic/FlowCoordinator+Juvenile.swift

```diff
@@ -35,37 +26,41 @@ public class JuvenileFlowCoordinator {
   /// - Parameters:
   ///   - completion: Always called at the end of the calls mentioned above on
   ///   the main queue.
-  public func startJuvenileFlow(completion: @escaping (Result<UINavigationController>) -> Void) {
-    guard let platformAPI = configuration.platformAPIInfo else {
-      let err = NSError(domain: ErrorDomain,
-                        code: ErrorCode.missingConfiguration.rawValue)
-      OperationQueue.main.addOperation {
-        completion(.fail(JuvenileFlowCoordinator.errorWithUserFriendlyMessage(amending: err)))
-      }
-      return
-    }
-
-    let config = configuration
-    let parentBarcode = configuration.juvenileParentBarcode
-
+  func startJuvenileFlow(completion: @escaping (Result<UINavigationController>) -> Void) {
     configuration.juvenileCreationHandler = { [weak self] juvenileInfo, responder in
-      self?.createPatron(using: platformAPI, parameters: juvenileInfo) { result in
+      guard let self = self else {
+        return
+      }
+      
+      self.createPatron(using: self.configuration.platformAPIInfo, parameters: juvenileInfo) { result in
         if case .fail(let err) = result {
-          responder?.handleJuvenilePatronResponse(.fail(
-            JuvenileFlowCoordinator.errorWithUserFriendlyMessage(amending: err)))
+          responder?.handleJuvenilePatronResponse(.fail(FlowCoordinator.errorWithUserFriendlyMessage(amending: err)))
           return
         }
 
         responder?.handleJuvenilePatronResponse(result)
       }
     }
 
-    checkJuvenileCreationEligibility(parentBarcode: parentBarcode) { (error) in
+    checkJuvenileCreationEligibility(parentBarcode: self.configuration.juvenileParentBarcode) { [weak self] error in
```

Comment:
I think there's a retain cycle here: 
self -> configuration (from line 30) -> closure -> self

I think that's why I had the 2 lines to pull out the juvenile parent barcode:
```
 let config = configuration
 let parentBarcode = configuration.juvenileParentBarcode
```